lib/capistrano/tasks/migrations.rake

```diff
@@ -0,0 +1,123 @@
+namespace :forkcms do
+  namespace :migrations do
+    desc 'Prepare the server for running Fork CMS migrations'
+    task :prepare do
+      on roles(:web) do
+        create_migrations_file
+      end
+    end
+
+    desc 'Execute pending migrations'
+    task :execute do
+      on roles(:web) do
+        # Prepare for migrations if it's not done already
+        Rake::Task["forkcms:migrations:prepare"].invoke()
+
+        # If the current symlink doesn't exist yet, execute the first migration
+        execute_initial_migration if not test "[[ -e #{current_path} ]]"
```

Comment:
Niet beter om te checken of de de file waarin de migrations worden bijgehouden bestaat?
Ik weet niet op welk punt de "current"-map wordt aangemaakt.

Reply:
De check voor de migrations file gebeurt daar vlak voor al (`Rake::Task["forkcms:migrations:prepare"].invoke()`

Nu ik eens nadenk over de logica van die 'initial migration'.. dat gaat niets doen. Ik geloof dat dat was om vroeger, bij het live zetten van een website ontwikkeld op staging, er voor te zorgen dat de database en de gegevens in `executed_migrations` gelijk liepen. Aangezien dat nu niet meer bestaat en `current` altijd al aangemaakt gaat zijn moet ik hier iets anders voor vinden.

Ik denk dat ik bij het aanmaken van de `executed_migrations` file al direct alle bestaande migrations ga moeten toevoegen. Tenzij er migrations zijn die nog niet uitgevoerd zijn op staging maar wel aanwezig zijn bij het voor het eerst deployen naar productie...

Het wordt nogal ingewikkeld denk ik, als je het niet begrijpt moet je maar eens tot hier komen om het irl te bespreken

Reply:
Ik zou er van uit gaan dat alle migrations al zijn uitgevoerd bij het aanmaken van de file.

Reply:
498f46b